spec/models/odk/response_parser_spec.rb

```diff
@@ -143,6 +144,17 @@
             end.to raise_error(SubmissionError, /Submission contains group or question/)
           end
         end
+
+        context "response is in wrong mission" do
```

Comment:
I'm not sure if this is testing anything interesting. The mission of Response gets assigned by CanCanCan in the controller so it's always going to be appropriate. What we really need to be testing is that if you submit a response with a form ID inside the XML that does not match the mission you're submitting it to, there should be a SubmissionError. Is that covered anywhere already?

The way to do this would to setup a response _without a form_, setup the XML with a form ID from a different mission, and then let it run. In a typical flow, the form won't be set on Response when it's passed into the parser. You might even try getting rid of the `||=` on line 173 of the parser and seeing if the specs still pass. I don't get why we have that there...